### PR TITLE
Add catalogue table

### LINKED ISSUE #2959: Create table for catalogue
Part of epic #2896 

## Is your feature request related to a problem? Please describe 👀
[Figma designs](https://www.figma.com/file/rIK2zVFUVhtbuXo0RBzOzw/0033-2305_omSupply-ColdChain-UI?type=design&node-id=2653-16424&mode=design&t=8KRUuAo7s1V0r0RM-0)

## Describe the solution you'd like 🎁
The [spreadsheet](https://docs.google.com/spreadsheets/d/1l1x38KT5cdt4Zd5piMWp3FuBuiSUv4faMHP8utDJDqU/edit?pli=1#gid=0) shows the likely columns, giving:

| column | type | null | notes |
|---|---|---|---|
| code | text | | |
| class_id | text | | FK on the reference data table |
| category_id | text | |  FK on the reference data table  |
| type_id | text | | | FK on the reference data table  |
| manufacturer | text | ✓ |  |
| model | text | | |
| catalogue | text | | Looking to only have `WHO PQS Catalogue` at this stage. even so, create an enum / custom data type for this in order to show the required info in the f/e |

I've taken the PQS catalogue and put into a [spreadsheet](https://docs.google.com/spreadsheets/d/1X3K2dEd6kBT0aG76KmhDYcGOvQMENb-oetbeZzHUGko/edit?usp=drive_link) though there is a more [complete spreadsheet](https://docs.google.com/spreadsheets/d/1jYgxg2xn6meHH77IztgRcyuUaSo7BecQ/edit#gid=1770418819) with additional lookup information

### Moneyworks Jobcode 🧰

`OMSCC:AST`

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #2959

# 👩🏻‍💻 What does this PR do? 
 
Adds the catalogue table.

# 🧪 How has/should this change been tested? 

## 📃 Documentation

_No user facing changes_

### CODE REVIEW

File: server/repository/src/migrations/v1_08_00/assets/mod.rs [hunk 1]
```diff
@@ -1,8 +1,10 @@
-use crate::StorageConnection;
+use crate::{assets::asset_category, StorageConnection};
 
+pub mod asset_catalogue;
 pub mod reference_data;
 
 pub(crate) fn migrate_assets(connection: &StorageConnection) -> anyhow::Result<()> {
     reference_data::migrate(connection)?;
+    asset_catalogue::migrate(connection)?;
```
Comment (mark-prins) [inline server/repository/src/migrations/v1_08_00/assets/mod.rs:8]: technically this is reference data as well.. but I thought it was nice to have in its own file

File: server/repository/src/migrations/v1_08_00/assets/asset_catalogue.rs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        r#"
+        CREATE TABLE asset_catalogue (
+            id TEXT NOT NULL PRIMARY KEY,
+            code TEXT NOT NULL,
+            asset_class_id TEXT NOT NULL REFERENCES asset_reference_data(id),
+            asset_category_id TEXT NOT NULL REFERENCES asset_reference_data(id),
+            asset_type_id TEXT NOT NULL REFERENCES asset_reference_data(id),
+            manufacturer TEXT,
+            model TEXT NOT NULL,
+            catalogue TEXT NOT NULL,
```
Comment (jmbrunskill) [inline server/repository/src/migrations/v1_08_00/assets/asset_catalogue.rs:15]: Kind of weird to me having a Catalogue column with in a catalog table...
Not convinced we need it. Let's discuss...
Maybe this table should be called `catalogue_item`?

**Review by jmbrunskill — APPROVED** [review]
Looks good, as discussed. I think it makes more sense to re-name to `catalogue_item`.
I don't think we need a `catalogue` table at this stage since there is just going to be 1...

Adding `id` filters for the reference data tables would allow frontend autocomplete.

File: server/repository/src/migrations/v1_08_00/assets/mod.rs (hunk 1 above)
Comment (mark-prins) [reply server/repository/src/migrations/v1_08_00/assets/mod.rs:8]: I think the others are fine to have together

File: server/repository/src/migrations/v1_08_00/assets/asset_catalogue.rs [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+pub(crate) fn migrate(connection: &StorageConnection) -> anyhow::Result<()> {
+    sql!(
+        connection,
+        r#"
+        CREATE TABLE asset_catalogue (
+            id TEXT NOT NULL PRIMARY KEY,
+            code TEXT NOT NULL,
+            asset_class_id TEXT NOT NULL REFERENCES asset_reference_data(id),
+            asset_category_id TEXT NOT NULL REFERENCES asset_reference_data(id),
+            asset_type_id TEXT NOT NULL REFERENCES asset_reference_data(id),
```
[the remark this answers is not in the document]
Comment (mark-prins) [reply server/repository/src/migrations/v1_08_00/assets/asset_catalogue.rs:12]: weird - I did have that corrected. must have overwritten 🙄

**Pull request merged**